### ISSUE TITLE
The bot is not saving expected tweet(s)

### ISSUE THREAD

**Issue by xaur** [JC60522/dcr_timestampbot#2]
tl;dr:

- Users expect the bot to save at least the *tweet they reply to*, but instead the bot is saving user's tweet that mentioned the bot, not the "target" tweet.

- Users expect the bot to save all tweets in the thread but it is only saving one tweet.

Now let's break it down. It looks like the users are confused about what to expect from the bot. This is a mix of two or three issues.

First let's revise what the bot is expected to do. From the [proposal](https://proposals.decred.org/record/0848cc9):

> In essence if you mention `@dcrtimestampbot` in a tweet your tweet will be timestamped on the Decred Blockchain and the tweet will be saved on IPFS. The bot will reply with a link to your downloadable timestamp proof as well as a link to view your tweet from IPFS via a dedicated domain + front end: https://dcr-timestampbot.com

This seems to be the way the bot actually works. But there is another description on the [bot website itself](https://dcr-timestampbot.com/):

> Timestamp tweets against the Decred blockchain using dcrtime and ipfs! Just reply to the tweet you want timestamped mentioning `@dcrtimestampbot`.
> 
> Every time `@dcrtimestampbot` is mentioned on twitter, a hash of the JSON-like string of the **entire thread** is timestamped and a file containing this string is stored on IPFS. 

This is not what the bot does, but it is what most users will expect after interacting with similar bots, e.g. the popular [Thread Reader App](https://threadreaderapp.com/) which saves entire threads.

Now a real example:

- An [interesting tweet](https://twitter.com/carolinecapital/status/1589287457975304193) is made
- A user [mentions the bot](https://twitter.com/elBisonMariscal/status/1590658837102530560) to save the tweet
- The bot [replies](https://twitter.com/dcrtimestampbot/status/1590658884305231874) "This **thread is stored** on IPFS" and gives two links.
- The [frontend link](https://dcr-timestampbot.com/ipfs/QmTgj8BwgD5BA4TK41V1mU4wK3fFuyXPehCgxD6udEZr9P) only shows the single tweet that triggered the bot, not the target tweet:

```
{
  "user": {"id_str": "1456600377932107781", "name": "El Mariscal \ud83e\uddac", "screen_name": "elBisonMariscal"},
  "id_str": "1590658837102530560",
  "created_at": "Thu Nov 10 10:53:14 +0000 2022",
  "text": "@carolinecapital @cz_binance @dcrtimestampbot lol",
  "repliedid": 1368514414505193475
}
```

In the Politeia proposal it was [confirmed](https://proposals.decred.org/record/0848cc9/comments/8) that the bot is only saving single tweets.

So in the end, the UI and/or the behavior is unexpected. At the very least the UI needs to be corrected to reflect the real behavior to expect. A complete fix however would be to implement what the bot advertises, since it is very useful and expected functionality. Archived *and* timestamped threads would be much stronger proofs than just archived threads made by the Thread Reader App.

**Comment by xaur** [comment]
Looking at the [latest timestamp](https://dcr-timestampbot.com/ipfs/QmcpnFha8jxXt8R35ENz53BVmtoFVv7zoQihXLLwYB11Km) I assume the expectation management part of this issue is resolved by updating the website info and the [README](https://github.com/JC60522/dcr_timestampbot/commit/66398833ce7f7ed677c73c54165337d5cb1b6ef9).

We'll need to wait for the next timestamp to verify that the tweet text template is updated as well.

The [templates](https://github.com/JC60522/dcr_timestampbot/tree/66398833ce7f7ed677c73c54165337d5cb1b6ef9/templates) in this repo still talk about threads, would you mind pushing the update to this repo please?

**Comment by JC60522** [comment]
> Looking at the [latest timestamp](https://dcr-timestampbot.com/ipfs/QmcpnFha8jxXt8R35ENz53BVmtoFVv7zoQihXLLwYB11Km) I assume the expectation management part of this issue is resolved by updating the website info and the [README](https://github.com/JC60522/dcr_timestampbot/commit/66398833ce7f7ed677c73c54165337d5cb1b6ef9).
> 
> We'll need to wait for the next timestamp to verify that the tweet text template is updated as well.
> 
> The [templates](https://github.com/JC60522/dcr_timestampbot/tree/66398833ce7f7ed677c73c54165337d5cb1b6ef9/templates) in this repo still talk about threads, would you mind pushing the update to this repo please?

Sure, no problem. Should be good now.

**Comment by JC60522** [comment]
Regarding this quoted concern:  **" A complete fix however would be to implement what the bot advertises, since it is very useful and expected functionality. Archived and timestamped threads would be much stronger proofs than just archived threads made by the Thread Reader App.**"

It's on my to-do list to find a solution for this. Once I have a structured plan together to resolve this issue I'll post a detailed suggestion here on how I propose to resolve this.

PS: I reopen this issue as I closed it by mistake on the previous reply.

**Comment by tiagoalvesdulce** [comment]
You can get tweets from a thread recursively because twitter API returns a `repliedid` field when the tweet is a reply to another tweet. If you can read JS code take a look at how I implemented it: https://github.com/tiagoalvesdulce/dcrtimestamptweet/blob/66603be202a05f8feb18687b8a29c1195db96a4f/helpers.js#L6

I used a pretty functional approach so should be pretty straight forward. Focus on `getThread `, `getThreadAux` and `getTweetById` functions.

Here is where the `getThread` is called: https://github.com/tiagoalvesdulce/dcrtimestamptweet/blob/66603be202a05f8feb18687b8a29c1195db96a4f/index.js#L31
